torch/fx/experimental/proxy_tensor.py

```diff
@@ -1311,6 +1311,7 @@ def __init__(
         from torch._inductor import config
 
         self.emulate_precision_casts = config.emulate_precision_casts
+        self.seen_invoke_subgraphs: Dict[str, torch.fx.GraphModule] = dict()
```

Comment:
I'm not sure how I feel about putting this on the ProxyMode. Some train of thoughts I want to go down:
- the cache is probably a per-compile thing, right? Can we put the cache in AOTAutograd?
- Can non-strict export compilation times benefit from invoke_subgraph? If so, where do we need to put the cache to make it work?

Reply:
1) There is torch._guards tracing_context if thats seems like a better place to consolidate all invoke_subgraph caches.
2) and 3) is primarily who is responsible for deduping the subgraphs. For simplicity and best compile time wins, I want this to be Dynamo responsibility. I agree that this will make Dynamo job harder.

Reply:
Yeah my general rule of thumb is: 
- if something goes in ProxyMode, it should be re-usable (as opposed to a one-time thing).
- If something goes in ProxyMode, it should be applicable to make_fx

Another question is: what is the user-facing API for invoke_subgraph? Does the user tag regions with key, or is that Dynamo's responsibility?

Follow-up question: do you think torch.compile should be able to de-duplicate invoke_subgraph invocations across torch.compile invocations? Like if we have the following user code (assuming the user doesn't tag regions with a key)
```py
invoke_subgraph(foo, x)
torch._dynamo.graph_break()
invoke_subgraph(foo, x)
```
Is Dynamo able to tag the regions with the same key?

Reply:
Yes, currently my preference is that Dynamo should be the one de-duping the same subgraphs.

When users call invoke_subgraph, they will use some helper function which will always generate a unique key. Sp, if you don't use Dynamo, then each invoke_subgraph call will be assumed to have different subgraphs.